axelor-contract/src/main/java/com/axelor/apps/contract/generator/InvoiceGeneratorContract.java

```diff
@@ -80,11 +79,25 @@ protected Invoice createInvoiceHeader() throws AxelorException {
       invoice.setInvoiceDate(appBaseService.getTodayDate(company));
     }
 
+    changeOperationType(contract, invoice);
+
     return invoice;
   }
 
   @Override
   public Invoice generate() throws AxelorException {
     return createInvoiceHeader();
   }
+
+  protected void changeOperationType(Contract contract, Invoice invoice) {
+    int targetTypeSelect = contract.getTargetTypeSelect();
+
+    if (targetTypeSelect == ContractRepository.YEB_CUSTOMER_CONTRACT) {
+      invoice.setOperationTypeSelect(InvoiceRepository.OPERATION_TYPE_CLIENT_REFUND);
+    }
+
+    if (targetTypeSelect == ContractRepository.YEB_SUPPLIER_CONTRACT) {
+      invoice.setOperationTypeSelect(InvoiceRepository.OPERATION_TYPE_SUPPLIER_REFUND);
+    }
+  }
```

Comment:
I don't understand why we are replacing the type at the end, and not change directly when the operationTypeSelect is set line 38-40.

Reply:
I tried to use a method in line 38-40 but it was not possible. But I fixed it by directly accessing the operationType attribute.